scripts-available/CDB_TableMetadata.sql

```diff
@@ -223,15 +214,39 @@ CREATE OR REPLACE FUNCTION public.CDB_TableMetadataTouch(tablename regclass)
     RETURNS void AS
     $$
     BEGIN
-        WITH upsert AS (
-            UPDATE public.cdb_tablemetadata
-            SET updated_at = NOW()
-            WHERE tabname = tablename
-            RETURNING *
+        WITH nv as (
+          SELECT tablename as tabname, NOW() as t
+        ), all_dependents as (
+          SELECT nv.tabname, nv.t
+          FROM nv
+          UNION
+          SELECT
+            dv.dependent_oid as tabname,
+            nv.t
+          FROM nv, public.CDB_TableMetadata_DependentViews(ARRAY[nv.tabname::regclass]) dv
+        ), updated as (
+          UPDATE public.CDB_TableMetadata x SET updated_at = ad.t
+          FROM (
+            SELECT
+              md.tabname,
+              ad.t
+            FROM public.CDB_TableMetadata md
+            JOIN all_dependents ad
+              ON ad.tabname = md.tabname
+            ORDER BY md.tabname ASC
+            FOR UPDATE OF md
+          ) ad
+          WHERE x.tabname = ad.tabname
+          RETURNING x.tabname
         )
-        INSERT INTO public.cdb_tablemetadata (tabname, updated_at)
-            SELECT tablename, NOW()
-            WHERE NOT EXISTS (SELECT * FROM upsert);
+        INSERT INTO public.CDB_TableMetadata SELECT ad.*
```

Comment:
Using CamelCase here made me think this was a function at first glance, aren't all tables in postgresql case insensitive (i.e: lowercase)?

Reply:
As are all functions in postgres! The function was using this casing before so I tried to maintain it.